WrongGearException.cpp

```diff
@@ -0,0 +1,11 @@
+#include "WrongGearException.hpp"
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+const char *WrongGearException::what() const noexcept { return msg.c_str(); }
```

Comment:
It's better not to override what() function, but pass a message to a proper constructor of a base class. You can use std::logic_error for example and you'll have it for free.